includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,36 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		$class_prefixes = [
+			'CalendarDay',
+			'CalendarMonth',
+			'CalendarMonthGrid',
+			'DayPicker',
+			'DayPickerKeyboardShortcuts',
+			'DayPickerNavigation',
+			'KeyboardShortcutRow',
+		];
+
+		if ( in_array( strtok( $class, '_' ), $class_prefixes, true ) ) {
+			return true;
+		}
+
+		return 'amp-date-picker-' === substr( $class, 0, 16 );
```

Comment:
Actually, is this not redundant with:

https://github.com/ampproject/amp-wp/blob/eea565f5a78d441428e049073354b40ae94d6399/includes/sanitizers/class-amp-style-sanitizer.php#L650-L655

Reply:
Good point, I think it is redundant.

What do you think of deleting that snippet you [shared](https://github.com/ampproject/amp-wp/pull/4339/files#r387407995)?


Otherwise, it might be confusing if that's there, and there's a call below of `->is_class_allowed_in_amp_date_picker()`

Reply:
I'd defer to @schlessera for his recommendation, since he highly optimized this method.

I'd probably rather change this method to:

```php
return in_array( strtok( $class, '_' ), $class_prefixes, true );
```

Reply:
Indeed, as explained in my comment above, the logic needs to be integrated into the switch statement that already has the datepicker included, otherwise this is getting a big performance hit again.

This would then take care of the `'amp-date-picker-' === substr( $class, 0, 16 )` bit, which would be a given once you enter here.

Keep in mind that this logic runs 20k to 30k times on a single request for a regular page. It is important to keep this highly optimized when making changes.

Reply:
@schlessera what do you think of this approach: b6f1a2360c84a82882f12c1f0f1126e4ee2a3450?

Reply:
@westonruter See https://github.com/ampproject/amp-wp/commit/b6f1a2360c84a82882f12c1f0f1126e4ee2a3450#r37853546